src/HealthChecks.AzureApplicationInsights/AzureApplicationInsightsHealthCheck.cs

```diff
@@ -0,0 +1,74 @@
+﻿using Microsoft.Extensions.Diagnostics.HealthChecks;
+using System.Runtime.ExceptionServices;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace HealthChecks.AzureApplicationInsights
+{
+    public class AzureApplicationInsightsHealthCheck : IHealthCheck
+    {
+        //from https://docs.microsoft.com/en-us/azure/azure-monitor/app/ip-addresses#outgoing-ports
+        private readonly string[] m_appInsightsUrls =
+        {
+            "https://dc.applicationinsights.azure.com",
+            "https://dc.applicationinsights.microsoft.com",
+            "https://dc.services.visualstudio.com"
+        };
+        private readonly string _instrumentationKey;
+
+        private readonly IHttpClientFactory _httpClientFactory;
+
+        public AzureApplicationInsightsHealthCheck(string instrumentationKey, IHttpClientFactory httpClientFactory)
+        {
+            _instrumentationKey = instrumentationKey ?? throw new ArgumentNullException(nameof(instrumentationKey));       
+            _httpClientFactory = httpClientFactory ?? throw new ArgumentNullException(nameof(httpClientFactory)); 
+        }
+
+        public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
+        {
+            try
+            {
+                return await CheckHealthAsyncInternal(context, cancellationToken);
+            }
+            catch (Exception ex)
+            {
+                return new HealthCheckResult(context.Registration.FailureStatus, exception: ex);
+            }
+
+            async Task<HealthCheckResult> CheckHealthAsyncInternal(HealthCheckContext context, CancellationToken cancellationToken = default)
```

Comment:
What is the purpose of that local function? Just inline it into `CheckHealthAsync`.

Reply:
you're totally right, I didn't think when I did that I'll fix it!